metomi/rosie/suite_id.py

```diff
@@ -456,6 +404,78 @@ def _from_location(self, location):
         if "commit:revision" in info_entry:
             self.revision = info_entry["commit:revision"]
 
+    @staticmethod
+    def _find_vc_file_from_location(location):
+        """Search a location and parents for a version control log file.
+
+        Args:
+            location: Path to search.
+
+        Returns:
+            If a file is found, a path, else None.
+        """
+        suite_engine_proc = SuiteEngineProcessor.get_processor()
+        suite_dir_rel_root = getattr(
+            suite_engine_proc, "SUITE_DIR_REL_ROOT", None
+        )
+        loc = Path(location).expanduser().resolve()
+        if suite_dir_rel_root:
+            sdrr = Path('~', suite_dir_rel_root).expanduser().resolve()
+            try:
+                loc.relative_to(sdrr)
+            except ValueError:
+                # Not an installed Cylc8 workflow run directory
+                pass
+            else:
+                # Slightly odd construction = loc + parents
+                for loc in (loc.relative_to(sdrr) / '_').parents:
+                    vcfilepath = sdrr / loc / SuiteId.VC_FILENAME
+                    if os.access(vcfilepath, os.F_OK | os.R_OK):
+                        return vcfilepath
+        return None
+
+    @staticmethod
+    def _parse_cylc_vc_file(fpath):
+        """Take a path to a Cylc VC file and returns an svn URL.
+
+        Args:
+            fpath: Location of Cylc Version Control log file.
+
+        Returns: SVN location, or None
+
+        Examples:
+            >>> class MockFpath():
+            ...     def read_text(self):
+            ...         return (
+            ...             'version control system="svn"\\n'
+            ...             'url="/a/b/c"\\n'
+            ...             'revision="4242"'
+            ...         )
+            >>> mypath = MockFpath()
+            >>> SuiteId.parse_cylc_vc_file(mypath)
+            '/a/b/c@4242'
+        """
+        location = None
+        vcsystem = None
+        url = None
+        rev = None
+        for line in fpath.read_text().split('\n'):
```

Comment:
Is it not possible to parse this using Rose's parser? I guess not because Rose doesn't allow multiline strings like
```
status = """
 M flow.cylc
 M whatever.txt
"""
```
?

Then perhaps it would make sense to save VCS info as JSON instead, as you suggested. But I think that would have to be a Cylc 8.0rc1 PR

Reply:
It's an internal which users shouldn't see: so IMO we should create an issue to change this beyond rc1. I don't think that it needs to be a blocker. 
If we want to be failsafe I can't think that it'd be massively expensive to write this + json both.